From 3c485aa9a8f8cb4171ec9b296fe407015b239ed6 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Thu, 8 Nov 2018 11:42:31 -0600
Subject: [PATCH] Skip more test activity if PB_SkipTests

Avoid test project restore and pack during source-build to remove prebuilts.
---
 build/CoreBuild.proj | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/build/CoreBuild.proj b/build/CoreBuild.proj
index 6e76d4fd..c6343231 100644
--- a/build/CoreBuild.proj
+++ b/build/CoreBuild.proj
@@ -25,7 +25,9 @@
         <ProjectReference Include="$(SrcDir)Microsoft.TemplateEngine.Edge\Microsoft.TemplateEngine.Edge.csproj"/>
         <ProjectReference Include="$(SrcDir)Microsoft.TemplateEngine.Orchestrator.RunnableProjects\Microsoft.TemplateEngine.Orchestrator.RunnableProjects.csproj"/>
         <ProjectReference Include="$(SrcDir)Microsoft.TemplateEngine.Utils\Microsoft.TemplateEngine.Utils.csproj"/>
+    </ItemGroup>
 
+    <ItemGroup Condition="'$(PB_SkipTests)' != 'true'">
         <ProjectReference Include="$(TestDir)Microsoft.TemplateEngine.Mocks\Microsoft.TemplateEngine.Mocks.csproj"/>
         <ProjectReference Include="$(TestDir)Microsoft.TemplateEngine.TestHelper\Microsoft.TemplateEngine.TestHelper.csproj"/>
     </ItemGroup>
@@ -66,14 +68,21 @@
                  Projects="$(BuildDir)Sign.csproj"
                  Targets="Restore" RunEachTargetSeparately="true" StopOnFirstFailure="true"/>
         <Message Importance="High" Text="Packing artifacts..." Condition="'$(SkipPack)' != 'true'"/>
-        <MSBuild Projects="$(SrcDir)Microsoft.TemplateEngine.Abstractions\Microsoft.TemplateEngine.Abstractions.csproj;
-                           $(SrcDir)Microsoft.TemplateEngine.Cli\Microsoft.TemplateEngine.Cli.csproj;
-                           $(SrcDir)Microsoft.TemplateEngine.Core\Microsoft.TemplateEngine.Core.csproj;
-                           $(SrcDir)Microsoft.TemplateEngine.Core.Contracts\Microsoft.TemplateEngine.Core.Contracts.csproj;
-                           $(SrcDir)Microsoft.TemplateEngine.Edge\Microsoft.TemplateEngine.Edge.csproj;
-                           $(SrcDir)Microsoft.TemplateEngine.Orchestrator.RunnableProjects\Microsoft.TemplateEngine.Orchestrator.RunnableProjects.csproj;
-                           $(SrcDir)Microsoft.TemplateEngine.Utils\Microsoft.TemplateEngine.Utils.csproj;
-                           $(TestDir)Microsoft.TemplateEngine.Mocks\Microsoft.TemplateEngine.Mocks.csproj"
+        <PropertyGroup>
+            <PackProjects>
+                $(SrcDir)Microsoft.TemplateEngine.Abstractions\Microsoft.TemplateEngine.Abstractions.csproj;
+                $(SrcDir)Microsoft.TemplateEngine.Cli\Microsoft.TemplateEngine.Cli.csproj;
+                $(SrcDir)Microsoft.TemplateEngine.Core\Microsoft.TemplateEngine.Core.csproj;
+                $(SrcDir)Microsoft.TemplateEngine.Core.Contracts\Microsoft.TemplateEngine.Core.Contracts.csproj;
+                $(SrcDir)Microsoft.TemplateEngine.Edge\Microsoft.TemplateEngine.Edge.csproj;
+                $(SrcDir)Microsoft.TemplateEngine.Orchestrator.RunnableProjects\Microsoft.TemplateEngine.Orchestrator.RunnableProjects.csproj;
+                $(SrcDir)Microsoft.TemplateEngine.Utils\Microsoft.TemplateEngine.Utils.csproj;
+            </PackProjects>
+            <PackProjects Condition="'$(PB_SkipTests)' != 'true'">
+                $(TestDir)Microsoft.TemplateEngine.Mocks\Microsoft.TemplateEngine.Mocks.csproj
+            </PackProjects>
+        </PropertyGroup>
+        <MSBuild Projects="$(PackProjects)"
                  Properties="PackageOutputPath=$(_PackageOutputPath);NoBuild=true;IncludeSymbols=true;PackSpecific=$(CoreOnly)"
                  Targets="Pack" StopOnFirstFailure="true" Condition="'$(SkipPack)' != 'true'"/>
 
-- 
2.17.1.windows.2

